### PR TITLE
Improve: advanced example of Backup and Restore

### PR DESCRIPTION
**Pull request by mianhk**
### What problem does this PR solve?

Improve advanced examples of Backup/BackupSchedule/Restore. Ref: #3279 

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [x] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Improve advanced examples of Backup/BackupSchedule/Restore.
```

### CODE REVIEW

File: examples/advanced/backup.yaml [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  ## - GCS_OBJECT_ACL
+  ## - GCS_BUCKET_ACL
+  ## - GCS_LOCATION
+  ## - GCS_STORAGE_CLASS
+  ## - GCS_SERVICE_ACCOUNT_JSON_KEY
+  ## - BR_LOG_TO_TERM
+  # env: []
+
+  ## From is the TidbCluster to be backed up.
+  from:
```
Comment (csuzhangxc) [inline examples/advanced/backup.yaml:37]: highlight `from` can be empty? ref #3443

File: examples/advanced/backup.yaml [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    ## TLSClientSecretName is the name of secret which stores tidb server client certificate.
+    ## Defaults to nil.
+    # tlsClientSecretName: ""
+  
+  ## BackupType is the backup type for TidbCluster.
+  ## `full` represents the full backup of TidbCluster.
+  ## `raw` represents the raw backup of TidbCluster.
+  ## `db` represents the backup of one DB of TidbCluster.
+  ## `table` represents the backup of one table of TidbCluster.
+  ## `tiflash-replica` represents restoring the tiflash replica removed by a failed restore of the older version BR.
```
Comment (csuzhangxc) [inline examples/advanced/backup.yaml:55]: this seems not valid now, ref https://github.com/pingcap/br/blob/c7f93829ef433dea4a86d0eb506a19a1339e4201/cmd/br/backup.go#L89-L92

https://github.com/pingcap/br/blob/c7f93829ef433dea4a86d0eb506a19a1339e4201/cmd/br/restore.go#L103-L107

File: examples/advanced/backup.yaml [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+  ## Defaults to Kubernetes default storage class.
+  # storageClassName: ""
+
+  ## StorageSize is the PV size specified for the backup operation.
+  ## This value must be greater than the size of the TidbCluster to be backed up.
+  ## Defaults to 100Gi.
+  storageSize: "100Gi"
+
+
+  ## BR configuration.
```
Comment (csuzhangxc) [inline examples/advanced/backup.yaml:158]: How about adding a link to the BR doc?

File: examples/advanced/backup.yaml [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+    # checksum: true
+    ## SendCredToTikv specifies whether the BR process passes its AWS or GCP privileges to the TiKV process.
+    ## Defaults to `true``.
+    # sendCredToTikv: true
+    ## OnLine specifies whether online during restore. Defaults to false.
+    # onLine: false
+    ## Options specifies the extra arguments that BR supports. These options has highest priority.
+    # options: []
+
+  ## DumplingConfig is the configs for dumpling
```
Comment (csuzhangxc) [inline examples/advanced/backup.yaml:193]: How about adding a link to the dumpling doc?

**Review by csuzhangxc — COMMENTED** [review]
some comments for `backup` also should be taken for `restore`

File: examples/advanced/backup.yaml (hunk 2 above)
Comment (mianhk) [reply examples/advanced/backup.yaml:55]: `backupType` is used in the controller now,  shall we modify this when we remove them from the controller?

File: examples/advanced/backup.yaml (hunk 4 above)
Comment (mianhk) [reply examples/advanced/backup.yaml:195]: The suggestions above have been fixed, PTAL.

File: examples/advanced/backup.yaml (hunk 2 above)
Comment (csuzhangxc) [reply examples/advanced/backup.yaml:56]: I mean, `tiflash-replica` is not valid in BR

File: examples/advanced/restore.yaml [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+  ## - GCS_OBJECT_ACL
+  ## - GCS_BUCKET_ACL
+  ## - GCS_LOCATION
+  ## - GCS_STORAGE_CLASS
+  ## - GCS_SERVICE_ACCOUNT_JSON_KEY
+  ## - BR_LOG_TO_TERM
+  # env: []
+
+  ## To is the TidbCluster to be restored.
+  to:
```
Comment (csuzhangxc) [inline examples/advanced/restore.yaml:37]: highlight `to` can be empty?

File: examples/advanced/backup.yaml [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+    ## Defaults to 10000.
+    pageSize: 10000
+    ## DisableBatchConcurrency specifies whether to disable the concurrent batch deletion method and uses the concurrent deletion method or not.
+    disableBatchConcurrency: false
+    ## BatchConcurrency specifies the number of goroutines to start for the concurrent batch deletion method.
+    ## Defaults to 10.
+    batchConcurrency: 10
+    ## RoutineConcurrency specifies the number of goroutines to start for the concurrent deletion method.
+    ## Defaults to 100.
+    routineConcurrency: 100
```
Comment (KanShiori) [inline examples/advanced/backup.yaml:256]: We can comment out these options,  most of the time users won't use them

File: examples/advanced/backup.yaml (hunk 2 above)
Comment (mianhk) [reply examples/advanced/backup.yaml]: Get it, fixed.

File: examples/advanced/restore.yaml (hunk 5 above)
Comment (mianhk) [reply examples/advanced/restore.yaml]: Yes, fixed.

**Review by csuzhangxc — APPROVED** [review]

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**